File: src/main/java/cloud/cosmin/checklister/dao/ListEntity.java
package cloud.cosmin.checklister.dao;

import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.List;
import java.util.Objects;
import java.util.UUID;

@Entity
@Table(name = "list")
public class ListEntity {
    @Id
    private UUID id;

    @Column(nullable = false)
    private String title;

    @OneToMany(mappedBy = "list")
    @Fetch(FetchMode.JOIN)
    private List<ItemEntity> items;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<ItemEntity> getItems() {
        return items;
    }

    public void setItems(List<ItemEntity> items) {
        this.items = items;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ListEntity that = (ListEntity) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(title, that.title) &&
                Objects.equals(items, that.items);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, title, items);
    }
}

File: src/main/kotlin/cloud/cosmin/checklister/dao/ListEntity.kt
package cloud.cosmin.checklister.dao

import org.hibernate.annotations.Fetch
import org.hibernate.annotations.FetchMode

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table
import java.util.Objects
import java.util.UUID

@Entity
@Table(name = "list")
class ListEntity {
    @Id
    var id: UUID? = null

    @Column(nullable = false)
    var title: String? = null

    @OneToMany(mappedBy = "list")
    @Fetch(FetchMode.JOIN)
    var items: List<ItemEntity>? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ListEntity?
        return id == that!!.id &&
                title == that.title &&
                items == that.items
    }

    override fun hashCode(): Int {
        return Objects.hash(id, title, items)
    }
}
